db/migrate/20160324174132_create_deployments.rb

```diff
@@ -0,0 +1,24 @@
+class CreateDeployments < ActiveRecord::Migration[5.0]
+  def change
+    create_table :container_deployments do |t|
+      t.string   :deployment_type
+      t.string   :deployment_version
+      t.boolean  :containerized, :default => true
+      t.string   :deployment_method
+      t.string   :deployed_provider_type
+      t.string   :deployed_provider_name
```

Comment:
I don't think I like storing these here.  The `_type` is not terrible, because that's not likely to change for the underlying provider.  But, the `_name` creates a housekeeping problem of having to know to update this record if ever the `deployed_provider`'s name changes.

In general, I think it's easy enough to just dereference the `deployed_ems.type` and `deployed_ems.name`.